### ISSUE TITLE
Report errors for `@Nullable` annotations on primitive / void locations?

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#662]
Such annotations used to be completely ignored by NullAway, but with recent changes they can lead to hard-to-understand errors.  E.g.:
```java
@Nullable int foo() { ... }
void baz(int i) { ... }
baz(foo());
```
For this code NullAway reports a warning of trying to pass `@Nullable` to a non-null parameter, whereas the real issue is the invalid `@Nullable` annotation on `foo()`'s return type.

We could directly check for such cases in NullAway and give a warning.  Alternately, Error Prone already has checkers around this type of issue:

https://errorprone.info/bugpattern/NullablePrimitive
https://errorprone.info/bugpattern/NullablePrimitiveArray
https://errorprone.info/bugpattern/NullableVoid

We could instead document that we recommend turning on those checkers in combination with NullAway.  That said, if we _really_ want to avoid giving weird error messages for these cases, we'll have to add more detection for primitive-type cases (which might slightly improve performance).
